src/Sets/Star.jl

```diff
@@ -0,0 +1,142 @@
+export Star,
+       center,
+       basis,
+       predicate
+
+"""
+    Star{N, VN<:AbstractVector{N}, MN<:AbstractMatrix{N}, PT<:AbstractPolyhedron{N}} <: AbstractStar{N}
+
+Type that represents a generalized star set where the predicate is polyhedral.
+
+### Fields
+
+- `c` -- vector that represents the center
+- `V` -- matrix where each column corresponds to a basis vector
+- `P` -- polyhedral set that represents the predicate
+
+### Notes
+
+The predicate function is implemented as a conjunction of linear constraints.
+This definition can be found in reference [1].
+
+### Examples
+
+This example is drawn from Example 1 in [1]. Consider the two-dimensional plane
+``\\mathbb{R}^2``. Let
+
+```jldoctest star_constructor
+julia> V = [[1.0, 0.0], [0.0, 1.0]]
+
+```
+be the set of unit vectors along the two axes, and take
+
+```jldoctest star_constructor
+julia> c = [3.0, 3.0]
+```
+be the center of the star set. Then, the star set `X = Star(c, V, P)` defines
+the set:
+
+```jldoctest star_constructor
+julia>
+```
+
+### References
+
+- [1] Duggirala, Parasara Sridhar, and Mahesh Viswanathan.
+      *Parsimonious, simulation based verification of linear systems.*
+      International Conference on Computer Aided Verification. Springer, Cham, 2016.
+"""
+struct Star{N, VN<:AbstractVector{N}, MN<:AbstractMatrix{N}, PT<:AbstractPolyhedron{N}} <: AbstractStar{N}
+    c::VN # center
+    V::MN # basis
+    P::PT # predicate
+    # TODO INNER CONSTRUCTOR WITH DIMENSION CHECK
+end
+
+# getter functions
+center(X::Star) = X.c
+basis(X::Star) = X.V
+predicate(X::Star) = X.P
+
+dim(X::Star) = length(X.c)
+
+function linear_map!(M::AbstractMatrix, X::Star, c′, V′)
```

Comment:
For linear map and affine map, I added alternative in-place implementations which do not allocate memory.